x-pack/plugins/stack_connectors/public/components/text_area_with_autocomplete.tsx

```diff
@@ -0,0 +1,348 @@
+        event.clientX > box.left &&
+        event.clientX < box.right &&
+        event.clientY > box.top &&
+        event.clientY < box.bottom
+      ) {
+        return;
+      }
+      closeList();
+    },
+    [closeList]
+  );
+
+  const onSelectMessageVariable = useCallback(
+    (variable: ActionVariable) => {
+      if (!textAreaRef.current) return;
+      const { selectionStart: startPosition, selectionEnd: endPosition } = textAreaRef.current;
+      const templatedVar = templateActionVariable(variable);
+
+      const newValue =
+        (inputTargetValue ?? '').substring(0, startPosition) +
+        templatedVar +
+        (inputTargetValue ?? '').substring(endPosition, (inputTargetValue ?? '').length);
+
+      editAction(paramsProperty, newValue, index);
+    },
+    [editAction, index, inputTargetValue, paramsProperty]
+  );
+
+  const renderSelectableOption = (option: any) => {
+    if (searchWord) {
+      return <EuiHighlight search={searchWord}>{option.label}</EuiHighlight>;
+    }
+    return option.label;
+  };
+
+  const selectableStyle: Properties<string | number> = useMemo(
+    () => ({
+      position: 'absolute',
+      top: popupPosition.top,
+      width: popupPosition.width,
+      left: popupPosition.left,
+      border: `${euiTheme.border.width.thin} solid ${euiTheme.border.color}`,
+      background: backgroundColor,
+      zIndex: euiThemeVars.euiZLevel1,
+    }),
+    [
+      backgroundColor,
+      euiTheme.border.color,
+      euiTheme.border.width.thin,
+      popupPosition.left,
+      popupPosition.top,
+      popupPosition.width,
+    ]
+  );
+
+  return (
+    <EuiFormRow
+      error={errors}
+      fullWidth
+      isDisabled={isDisabled}
+      isInvalid={errors && errors.length > 0 && inputTargetValue !== undefined}
+      label={label}
+      labelAppend={
+        <AddMessageVariables
+          messageVariables={messageVariables}
+          onSelectEventHandler={onSelectMessageVariable}
+          paramsProperty={paramsProperty}
+        />
+      }
+    >
+      <>
+        <EuiOutsideClickDetector onOutsideClick={clickOutSideTextArea}>
+          <EuiTextArea
+            disabled={isDisabled}
+            inputRef={textAreaRef}
+            fullWidth
+            isInvalid={errors && errors.length > 0 && inputTargetValue !== undefined}
+            name={paramsProperty}
+            value={inputTargetValue || ''}
+            data-test-subj={`${paramsProperty}TextArea`}
+            onChange={onChangeWithMessageVariable}
+            onFocus={useCallback(() => setListOpen(true), [])}
+            onKeyDown={textareaOnKeyPress}
+            onBlur={useCallback(() => {
+              if (!inputTargetValue && !isListOpen) {
+                editAction(paramsProperty, '', index);
+              }
+            }, [editAction, index, inputTargetValue, isListOpen, paramsProperty])}
+            onClick={useCallback(() => closeList(), [closeList])}
+            onScroll={useCallback(() => {
```

Comment:
I've try to fix it fast. Did not work, because when our main logic will be broken and autocomplete will work with bugs.
I've created separate task for that: https://github.com/elastic/kibana/issues/165204.
For this PR think that it's fine to keep it like this, because we have similar behaviour in different places in rule form.
Are you ok with that?

Reply:
we were able to figure it out check the last commit